lib/publiccloud/instance.pm

```diff
@@ -433,7 +447,7 @@ sub softreboot {
 
     # wait till ssh disappear
     while (($duration = time() - $start_time) < $args{timeout}) {
-        last unless (defined($self->wait_for_ssh(timeout => 1, proceed_on_failure => 1, username => $args{username})));
+        last unless (defined($self->wait_for_ssh(timeout => 1, proceed_on_failure => 1, username => $args{username}, systemup_check => 0)));
```

Comment:
This doesn't make sense to me.

Reply:
Reading the comment in L448, we can see that it is important instead, for the `stop` phase.
During _stopping_ phase of a `soft reboot` we have the reverse process than the access and ssh is the last service that we can interrogate and will no more answer, while during the start it will be the first one to answer. 
It is like when you're exiting from a room, waiting until the port is closed.

Reply:
Reading the comment in L448, we can see that it is important instead, for the `stop` phase.
During _stopping_ phase of a `soft reboot` we have the reverse process than the access and ssh is the last service that we can interrogate and will no more answer when stop completed, therefore we have also `proceed_on_failure => 1`,but no need to check also the system up; during the start phase instead, ssh will be the first one to answer. 
It is like when a party, beers, music, all finished and you're exiting from that house: after that the door (ssh) is closed, you guess that party is over.

Reply:
Reading the comment in L448, we can see that it is important instead, for the `stop` phase.
During _stopping_ phase of a `soft reboot` we have the reverse process than the start and access and ssh is the last service that we can interrogate and will no more answer when stop completed, therefore we have also `proceed_on_failure => 1`,but no need to check also the system up; during the start phase instead, ssh will be the first one to answer. 
It is like when a party, beers, music, all finished and you're exiting from that house: after that the door (ssh) is closed, you guess that party is over.

Reply:
I see, we call `$self->wait_for_ssh()` below on line 454.